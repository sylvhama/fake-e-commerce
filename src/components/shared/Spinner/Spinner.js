import styled, { keyframes } from 'styled-components';
import PropTypes from 'prop-types';

const rotate360 = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

const Spinner = styled.span`
  border: ${props => props.borderWidth}px solid rgba(0, 0, 0, 0.2);
  border-top: ${props => props.borderWidth}px solid
    ${props => props.theme.color.primary};
  width: ${props => props.size}px;
  height: ${props => props.size}px;
  display: inline-block;
  border-radius: 50%;
  animation: ${rotate360} 1s linear infinite;

  ${props =>
    props.block &&
    `
      margin: ${props.theme.spacing.reg} auto;
      display: block;
  `};
`;

Spinner.propTypes = {
  size: PropTypes.number,
  borderWidth: PropTypes.number,
  block: PropTypes.bool
};

Spinner.defaultProps = {
  size: 32,
  borderWidth: 4
};

export default Spinner;
