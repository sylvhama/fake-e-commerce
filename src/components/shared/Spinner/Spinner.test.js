import React from 'react';
import 'jest-styled-components';

import { renderWithTheme } from '../../../tests/helpers';

import Spinner from './Spinner';

test('Spinner renders correctly', () => {
  const rendered = renderWithTheme(<Spinner />).toJSON();
  expect(rendered).toMatchSnapshot();
});

test('Spinner renders correctly with block', () => {
  const rendered = renderWithTheme(<Spinner block />).toJSON();
  expect(rendered).toMatchSnapshot();
  expect(rendered).toHaveStyleRule('display', 'block');
  expect(rendered).toHaveStyleRule('margin', '16px auto');
});

test('Spinner renders correctly specific size and border width', () => {
  const rendered = renderWithTheme(
    <Spinner size={42} borderWidth={10} />
  ).toJSON();
  expect(rendered).toMatchSnapshot();
  expect(rendered).toHaveStyleRule('width', '42px');
  expect(rendered).toHaveStyleRule('height', '42px');
  expect(rendered).toHaveStyleRule('border', '10px solid rgba(0,0,0,0.2)');
});
