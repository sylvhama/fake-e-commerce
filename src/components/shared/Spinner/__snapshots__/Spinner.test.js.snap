// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders correctly 1`] = `
.c0 {
  border: 4px solid rgba(0,0,0,0.2);
  border-top: 4px solid #ea352d;
  width: 32px;
  height: 32px;
  display: inline-block;
  border-radius: 50%;
  -webkit-animation: iVXCSc 1s linear infinite;
  animation: iVXCSc 1s linear infinite;
}

<span
  className="c0"
  size={32}
/>
`;

exports[`Spinner renders correctly specific size and border width 1`] = `
.c0 {
  border: 10px solid rgba(0,0,0,0.2);
  border-top: 10px solid #ea352d;
  width: 42px;
  height: 42px;
  display: inline-block;
  border-radius: 50%;
  -webkit-animation: iVXCSc 1s linear infinite;
  animation: iVXCSc 1s linear infinite;
}

<span
  className="c0"
  size={42}
/>
`;

exports[`Spinner renders correctly with block 1`] = `
.c0 {
  border: 4px solid rgba(0,0,0,0.2);
  border-top: 4px solid #ea352d;
  width: 32px;
  height: 32px;
  display: inline-block;
  border-radius: 50%;
  -webkit-animation: iVXCSc 1s linear infinite;
  animation: iVXCSc 1s linear infinite;
  margin: 16px auto;
  display: block;
}

<span
  className="c0"
  size={32}
/>
`;
